addon/usables/use-machine.js

```diff
@@ -69,15 +69,34 @@ export class MachineInterpreterManager {
     interpreter.setup();
   }
 
-  updateUsable() {
-    // not sure if it makes sense to support this - issue warning
-    warn(ARGS_STATE_CHANGE_WARNING, false, { id: 'statecharts.use-machine.args-state-change' });
+  updateUsable(bucket, configurableMachineDefinition) {
+    const { interpreter } = bucket;
+    const { args, _update } = configurableMachineDefinition;
+
+    if (_update) {
+      const { machine, context, config } = args;
+      _update({
+        machine,
+        context,
+        config,
+        send: interpreter.service.send,
+        restart: this.restartUsable.bind(this, bucket, configurableMachineDefinition),
+      });
+    } else {
+      warn(ARGS_STATE_CHANGE_WARNING, false, { id: 'statecharts.use-machine.args-state-change' });
+    }
   }
 
   teardownUsable({ interpreter }) {
     // teardown interpreter
     interpreter.teardown();
   }
+
+  restartUsable(bucket, configurableMachineDefinition) {
```

Comment:
Is it really necessary to go so far as teardown and setup all over again? Xstate has stop and start methods that return to the initial state: https://xstate.js.org/docs/guides/interpretation.html#interpreter

Reply:
Do you see any problems with a  teardown/setup? We want to restart the usable here because people can also use `useMachine` with a dynamic machine definition:

Example:
```
@use statechart = createMachine(this.args.machine)
```

i.e. we can't just stop and start the interpreter because that might not be enough. we want to run through the entire setup again because basically anything can change.

Reply:
I think `interpreter.stop().start()` wouldn't work if the context changes as well...